x/rollapp/transfergenesis/ibc_module.go

```diff
@@ -107,39 +109,32 @@ func (w IBCModule) OnRecvPacket(
 	}
 
 	if !transfer.IsRollapp() {
-		l.Debug("Transfer is not from a rollapp.")
 		return w.IBCModule.OnRecvPacket(ctx, packet, relayer)
 	}
 
+	ra := transfer.Rollapp
+
 	memo, err := getMemo(transfer.GetMemo())
 	if errorsmod.IsOf(err, gerr.ErrNotFound) {
-		// If someone tries to send a transfer without the memo before the bridge is open, they will
-		// be blocked at the transfersenabled middleware
+		// The first regular transfer marks the full opening of the bridge, more genesis transfers will not be allowed.
+		w.rollappKeeper.EnableTransfers(ctx, ra.RollappId)
 		return w.IBCModule.OnRecvPacket(ctx, packet, relayer)
 	}
 	if err != nil {
 		l.Error("Get memo.", "err", err)
 		return channeltypes.NewErrorAcknowledgement(errorsmod.Wrap(err, "get memo"))
 	}
 
-	ra := transfer.Rollapp
-
-	nTransfersDone, err := w.rollappKeeper.VerifyAndRecordGenesisTransfer(ctx, ra.RollappId, memo.TotalNumTransfers)
-	if errorsmod.IsOf(err, derr.ErrViolatesDymensionRollappStandard) {
-		// The rollapp has deviated from the protocol!
-		handleFraudErr := w.handleFraud(ra.RollappId)
+	if ra.GenesisState.TransfersEnabled {
+		// Genesis transfers are disabled once the bridge is already open
+		err = w.handleFraud(ctx, ra.RollappId)
```

Comment:
need to make sure we don't create DoS attack vector.
in current implementation, attacker who manages to submit "genesistransfer" memo, will DoS the rollapp

Reply:
this is already blocked on the RA
this was foreseen in the original ADR

Reply:
basically if a rollapp allows this it doesn't conform to DRS

Reply:
well bugs are bugs
e.g we don't wont that RA packet sent on height 1_000_000 will DoS the rollapp because of genesis transfer feature which is far in history
It same concept as invariant. Not for every thing we want to have invariables. only for those that actually fkc up something.
if "genesis transfer" memo is sent on height 1_000_000 it should raise a lot of alarms, but not sure it need to fraud the rollapp.

Reply:
I've anyways renamed fraud to drs violation here